packages/xchain-ethereum/src/client.ts

```diff
@@ -1,85 +1,148 @@
-  private _network: Network
-  private _phrase: Phrase
+  private _network: EthNetwork
+  private _phrase: string
   private _provider: Provider
   private _address: Address
   private _etherscan: EtherscanProvider
-  private _vault: ethers.Contract | null = null
-
-  constructor(network: Network = Network.TEST, phrase?: Phrase, vault?: string) {
-    if (phrase && !validateMnemonic(phrase)) {
+  private _vault?: Address
+  private blockchairNodeUrl = ''
+  private blockchairNodeApiKey = ''
+
+  constructor({
+    network = 'testnet',
+    blockchairUrl = '',
+    blockchairNodeApiKey = '',
+    phrase,
+    vault,
+    etherscanApiKey,
+  }: ClientParams) {
+    if (phrase && !Crypto.validatePhrase(phrase)) {
       throw new Error('Invalid Phrase')
     } else {
-      this._phrase = phrase || generateMnemonic()
-      this._network = network
-      this._provider = getDefaultProvider(network)
+      this._phrase = phrase || Crypto.generatePhrase() // @todo do we need to set a size here?
+      this._network = xchainNetworkToEths(network)
+      this._provider = getDefaultProvider(this._network)
       this._wallet = ethers.Wallet.fromMnemonic(this._phrase)
       this._address = this._wallet.address
-      this._etherscan = new EtherscanProvider(this._network) // for tx history
-      if (vault) this.setVault(vault)
+      this._vault = vault
+      this._etherscan = new EtherscanProvider(this._network, etherscanApiKey) // for tx history
+
       // Connects to the ethereum network with it
-      const provider = getDefaultProvider(this._network)
-      const newWallet = this.wallet.connect(provider)
-      this.changeWallet(newWallet)
+      this.changeWallet(this.wallet.connect(this._provider))
+      this.setBlockchairNodeURL(blockchairUrl)
+      this.setBlockchairNodeAPIKey(blockchairNodeApiKey)
     }
   }
 
+  purgeClient(): void {
+    this.setPhrase('')
+  }
+
+  getNetwork(): XChainNetwork {
+    return ethNetworkToXchains(this._network)
+  }
+
+  setBlockchairNodeURL = (url: string): void => {
+    this.blockchairNodeUrl = url
+  }
+
+  setBlockchairNodeAPIKey(key: string): void {
+    this.blockchairNodeApiKey = key
+  }
+
   /**
    * Getters
    */
   getAddress(): Address {
     return this._address
   }
 
-  protected getExplorerUrl = (): string => {
-    return ''
+  getVault(): Address | undefined {
+    return this._vault
   }
 
-  get wallet(): ethers.Wallet {
-    return this._wallet
+  // @todo what url ?
+  getExplorerUrl = (): string => {
+    return ''
   }
 
-  get vault(): ethers.Contract | null {
-    return this._vault
+  // @todo what url ?
+  getExplorerAddressUrl = (address: Address): string => {
+    return `${this.getExplorerUrl()}/address/${address}`
+  }
+  // @todo what url ?
+  getExplorerTxUrl = (txID: string): string => {
+    return `${this.getExplorerUrl()}/tx/${txID}`
   }
 
-  get network(): Network {
-    return this._network
+  get wallet(): ethers.Wallet {
+    return this._wallet
```

Comment:
Rename all `get` (getter)  to `getXYZ` to be consistent with other clients.

BTW: What's the reason to make `wallet` public?